[FE] Issue Detail Page API 연동

moaikang opened the pull request:
### 새로운 기능

- Issue Detail Page API 연동

### 작업 유형
- [x] 신규 기능 추가
- [ ] 버그 수정
- [ ] 리펙토링
- [ ] 문서 업데이트

### 체크리스트
- [x] Merge 하는 브랜치가 올바른가?
- [x] 코딩컨벤션을 준수하는가?
- [x] PR과 관련없는 변경사항이 없는가?
- [ ] 내 코드에 대한 자기 검토가 되었는가?
- [ ] 변경사항이 효과적이거나 동작이 작동한다는 것을 보증하는 테스트를 추가하였는가?
- [ ] 새로운 테스트와 기존의 테스트가 변경사항에 대해 만족하는가?

Commit:
feat    : :zap:️ Issue Detail context

issue-detail-page의 상태를 context로 옮겨보았다.

Commit:
feat    : :zap:️ Context API에 get Issue 연동 후 적용

Context API에 getIssueById 연동후에 적용하였다.

Commit:
feat    : :zap:️ issue-Detail-page에 필요한 API 연동

issue-Detail-page에 필요한 API를 연동하였다.

Commit:
feat    : :zap:️ 연동한 API에 맞게 component 수정

연동한 API에 맞게 component의 데이터를 뿌려주게 수정하였다.

Commit:
style   : :high_heel: console.log 제거

미처 찾지 못한 console.log를 제거하였다.

Commit:
style   : :high_heel: 함수 분리

onClick 같은 이벤트 발생 함수에 들어가 있는 함수의 로직이 복잡할 때, 다른 함수로 분리하였다.

Commit:
fix     : :fire: 존재하지 않는 id의 파라미터로 접근했을 때

존재하지 않는 id의 파라미터로 접근했을 때 존재하지 않는 id로 접근했다고 렌더링되게 예외처리 하였다.

Commit:
fix     : :fire: 마일스톤이 없을 경우에 예외 처리

issueDetailPage에 마일스톤이 없을 경우에 예외 처리를 진행하였다.

maong0927 commented on web/client/src/components/issue-sidebar/sidebar/index.tsx:
```diff
@@ -36,6 +36,7 @@ export default function IssueSideBar({
   labels,
   milestone,
 }: IssueSideBarProps) {
+  console.log(milestone);
```
완벽한 그의 작은 실수 ㅠ

moaikang replied:
수정했습니다..!

Commit:
style   : :high_heel: console 삭제

console.log를 삭제하였다.

maong0927 reviewed (approved):
수고하셨습니다!!

namda-on commented on web/client/src/components/issueHeader/index.tsx:
```diff
@@ -20,38 +24,84 @@ export default function IssueHeader({
+    }
+  };
+
+  const onClickUpdateEditBox = async () => {
+    const result = await api.updateIssueTitle(id, editBoxValue);
+    setIsTitleUpdate(true);
+    toggleEditBox();
+  };
```
result 에서 서버에서 응답이 ok 인지 확인하시고  `setIsTitleUpdate(true);`  를 실행하면 더 좋을 것 같습니다! 
db에 반영안되고 랜더링만 바뀔수도 있을 것 같습니다!

moaikang replied:
오..! 감사합니다 미처 생각하지 못했어요!

namda-on commented on web/client/src/lib/api.ts:
```diff
@@ -48,15 +48,15 @@ interface CommentProps {
   })
 
-  if(!res.ok) return;
+  if(!res.ok) return false;
   return await res.json();
 }
```
오 그냥 return 하는게 아니라 false를 보내주는 것도 좋겠네요! 정말 좋습니당

namda-on reviewed (approved):
고생하셨습니다! ㅎㅎ 많이 배워갑니다 총총..

Commit:
fix     : :fire: onClickUpdateEditBox() 리팩토링

result 에서 서버에서 응답이 ok 인지 확인하지 않고 setIsTitleUpdate(true)
db에 반영안되고 랜더링만 바뀔수도 있을 것 같다는 리뷰를 반영해 코드를 수정하였다.

Commit:
fix     : :fire: 방금 전 커밋 수정

방금 전 커밋에서 반영하지 못한 커밋을 추가하였습니다.
